lib/winapi/thread.c

```diff
@@ -1,7 +1,64 @@
+#include <assert.h>
+#include <string.h>
 #include <processthreadsapi.h>
 #include <winbase.h>
+#include <pdclib/_PDCLIB_xbox_tls.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+VOID NTAPI WinapiThreadStartup (PKSTART_ROUTINE StartRoutine, PVOID StartContext)
+{
+    // Make sure the TLS pointer is valid
+    assert(KeGetCurrentThread()->TlsData);
+
+    // Retrieve TLS area address and point first entry to itself
+    DWORD *TlsData;
+    TlsData = ((DWORD *)KeGetCurrentThread()->TlsData) + 1;
+    TlsData[-1] = (DWORD)TlsData;
+
+    // 16 byte alignment required
+    assert(((DWORD)TlsData & 15) == 0);
+
+    // Copy initial values of TLS area
+    DWORD TlsDataSize;
+    TlsDataSize = _tls_used.EndAddressOfRawData - _tls_used.StartAddressOfRawData;
+    memcpy(TlsData, (void *)_tls_used.StartAddressOfRawData, TlsDataSize);
+
+    // Zero-initialize the rest
+    RtlZeroMemory((char *)TlsData + TlsDataSize, _tls_used.SizeOfZeroFill);
+
+    int res;
+    res = (*(LPTHREAD_START_ROUTINE)StartRoutine)(StartContext);
+    PsTerminateSystemThread(res);
+}
+
+HANDLE CreateThread (LPSECURITY_ATTRIBUTES lpThreadAttributes, SIZE_T dwStackSize, LPTHREAD_START_ROUTINE lpStartAddress, LPVOID lpParameter, DWORD dwCreationFlags, LPDWORD lpThreadId)
```

Comment:
`assert(lpThreadAttributes == NULL)` ?
Maybe even return error for invalid arguments?
At least there should be a comment about this not being supported.

Reply:
I added a comment, but I'd like to keep ignoring the parameter - using it doesn't make sense on the Xbox, but there may be code out there using it. Ignoring it should be fine for conforming to MSDN descriptions.